protozoo/predictor.py

```diff
@@ -1,23 +1,34 @@
 import torch
+
 from ignite.engine import Events, Engine
+from typing import Optional
 
 from protozoo.tiktorch_config_keys import ModelZooEntry
 
 
-def get_predictor(model_zoo_entry: ModelZooEntry) -> Engine:
-    def predict_batch(predictor: Engine, batch) -> torch.Tensor:
-        print('predict')
-        return predictor.state.model(batch)
+class Predictor(Engine):
+    def __init__(self, model_zoo_entry: ModelZooEntry, model: Optional[torch.nn.Module] = None):
+        if model_zoo_entry.model_config.pretrained_source is not None:
+            raise NotImplementedError("model_zoo_entry.model_config.pretrained_source")
+
+        model_config = model_zoo_entry.model_config
 
+        def predict_batch(predictor: Engine, batch) -> torch.Tensor:
+            print("predict")
+            return predictor.state.model(batch)
```

Comment:
Why is `predict_batch` inside of the `init`?

Reply:
Ok, I get it.

Reply:
I wonder if it's not smarter to move this out of the init.
This would allow to override `predict_batch` in a deriving class as a way to modify the behaviour.

Reply:
discussion turned into coding: https://github.com/kreshuklab/protozoo/pull/1/commits/85a6c813b0ce3fb662353a48574f53854a6e4c3e